lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -344,9 +344,16 @@ protected function fetchVersionValue($versionedClass, array $id)
              . ' FROM '  . $tableName
              . ' WHERE ' . implode(' = ? AND ', $identifier) . ' = ?';
 
+        $types = [];
+        foreach ($id as $field => $value) {
+            foreach ($this->getTypes($field, $value, $versionedClass) as $type) {
```

Comment:
Do we need to iterate this too? I mean in theory we could use `array_merge()` right, like:

```php
$types = [];

foreach ($id as $field => $value) {
    $types = array_merge($types, $this->getTypes($field, $value, $versionedClass));
}
```